lib/IRGen/GenDecl.cpp

```diff
@@ -2789,6 +2862,11 @@ llvm::Function *IRGenModule::getAddrOfSILFunction(
   if (auto clangDecl = f->getClangDecl()) {
     auto globalDecl = getClangGlobalDeclForFunction(clangDecl);
     clangAddr = getAddrOfClangGlobalDecl(globalDecl, forDefinition);
+
+    if (auto ctor = dyn_cast<clang::CXXConstructorDecl>(clangDecl)) {
+      clangAddr =
+          emitCXXConstructorThunkIfNeeded(*this, f, ctor, entity, clangAddr);
```

Comment:
Hmm, so we're always going through the thunk?  Alright, I guess this is okay.

Reply:
`emitCXXConstructorThunkIfNeeded` will exit early (returning `clangAddr`) if the thunk isn't needed.

Reply:
Sure, that's not quite what I meant but it's okay.

Reply:
OK. If you want me to change something here, just let me know.

Reply:
No, it's fine, I don't think there's any reason for us to do extra work to try to always do a direct call.

Reply:
I agree, at least for now. Maybe in the future. I think it would be doable.